src/connectionimpl.cpp

```diff
@@ -123,7 +123,9 @@ uint64_t ConnectionImpl::parse(const Buffer &buffer)
         try
         {
             // try to recognize the frame
-            ReceivedFrame receivedFrame(ReducedBuffer(buffer, processed), _maxFrame);
```

Comment:
> Why do you think that It is an undefined behaviour?

The next statement is:
`
if (receivedFrame.complete())
`
ReducedBuffer is destroyed here because that **full expression containing the call** is already evaluated.
But ReceivedFrame::complete uses stored reference to the destroyed ReducedBuffer.
`
bool ReceivedFrame::complete() const
{
    return _buffer.size() >= _payloadSize + 8;
}
`